database/migration/src/main/scala/cromwell/database/migration/custom/BatchedTaskChange.scala

```diff
@@ -0,0 +1,140 @@
+package cromwell.database.migration.custom
+
+import java.sql.{PreparedStatement, ResultSet}
+
+import cromwell.database.migration.metadata.table.symbol.{QueryPaginator, ResultSetIterator}
+import liquibase.database.jvm.JdbcConnection
+import liquibase.exception.CustomChangeException
+
+/**
+  * Runs a migration as a series of batches.
+  */
+trait BatchedTaskChange extends MigrationTaskChange {
+  /**
+    * Returns the maximum primary key for the table.
+    *
+    * Example:
+    * {{{
+    *   SELECT MAX([PRIMARY_KEY])
+    *   FROM [TABLE];
+    * }}}
+    */
+  def readCountQuery: String
+
+  /**
+    * Returns a statement that returns rows passed to migrateBatchRow, batching on a primary key between the half-open
+    * primary key range [start, stop).
+    *
+    * Example:
+    * {{{
+    *   SELECT [COLUMNS]
+    *   FROM [TABLE]
+    *   WHERE [PRIMARY_KEY] >= ? AND [PRIMARY_KEY] < ?;
+    * }}}
+    */
+  def readBatchQuery: String
+
+  /**
+    * Used to prepare the statement that will be passed repeatedly into migrateRow.
+    *
+    * Example:
+    * {{{
+    *   UPDATE [TABLE]
+    *   SET [COLUMNS]
+    *   WHERE [PRIMARY_KEY] = ?;
+    * }}}
+    *
+    * Example:
+    * {{{
+    *   INSERT INTO [TABLE]
+    *   SET [COLUMNS];
+    * }}}
+    */
+  def migrateBatchQuery: String
+
+  /**
+    * Migrate a row.
+    *
+    * Read the values from readRow, update the values, set the updated values on the migrateStatement, and then call
+    * migrateStatement.addBatch(). Return the (estimated) number of rows to be written by this batch.
+    *
+    * @param readRow          The row to migrate
```

Comment:
I thought about changing the name to better match the type (this code is copied from elsewhere). But I like that the name conveys that the implementor _is going to have a bad time_ if they call any other methods on the `ResultSet` besides accessing the current row values.